packages/lwc/types.d.ts

```diff
@@ -7,6 +7,8 @@
 /**
  * Lightning Web Components core module
  */
+import { WireAdapterConstructor } from '../@lwc/engine-core/src/framework/wiring';
```

Comment:
This will not work when users install the `lwc` package because it there is no direct dependency on `@lwc/engine-core`. There's no transitive dependency either--if you take a look at this package's `@lwc/engine` dependency, you'll notice that it's just a placeholder package that copies files over from `@lwc/engine-dom` during build time.